### PR TITLE
Add attribute to allow csharp compilation of unsafe code

### LINKED ISSUE #352: Support /unsafe compiler flag
I am trying to figure out how to set custom compiler flags for a `csharp_binary`. This is my first time using bazel in a project (actually porting an existing project just to learn) I would like to know if it is possible or if it needs to be implemented.

The error I am referring to is `error CS0227: Unsafe code may only appear if compiling with /unsafe`

### PR DESCRIPTION
**Pull request by fedemartinezdev**
Provides a way to enable unsafe compilation for C#

```
csharp_binary(
   name = "mybinary",
   ...,
   allow_unsafe = True
)
```

Closes #352

### CODE REVIEW

File: dotnet/private/rules/csharp/actions/csharp_assembly.bzl [hunk 1]
```diff
@@ -303,13 +308,16 @@ def _compile(
...
         warning_level,
+        allow_unsafe,
         out_dll = None,
         out_ref = None,
         out_pdb = None):
     # Our goal is to match msbuild as much as reasonable
     # https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/listed-alphabetically
     args = actions.args()
     args.add("/unsafe-")
+    if (allow_unsafe):
```
Comment (purkhusid) [inline dotnet/private/rules/csharp/actions/csharp_assembly.bzl:319]: Could you change this to:
```
if allow_unsafe:
    args.add("/unsafe+")
else:
   args.add("/unsafe-")
```

File: dotnet/private/rules/common/attrs.bzl [hunk 2]
```diff
@@ -130,6 +130,11 @@ COMMON_ATTRS = {
         doc = "Whether or not to override the strict_deps attribute.",
         mandatory = False,
     ),
+    "allow_unsafe": attr.bool(
```
Comment (purkhusid) [inline dotnet/private/rules/common/attrs.bzl:133]: Could you move this to `CSHARP_COMMON_ATTRS` instead since `unsafe` is not available in F#

**Review by purkhusid — COMMENTED** [review]
Thanks for your contribution! Couple of comments and I would really appreciate if you could add a very simple test under `dotnet/private/tests/unsafe` that compiles something that is using `unsafe.`

**Comment by purkhusid** [comment]
Ah, also just noticed the WIP in the title. Let me know when you think it's ready for review.

**Comment by fedemartinezdev** [comment]
@purkhusid I think it is ready!

File: dotnet/private/rules/common/attrs.bzl [hunk 3]
```diff
@@ -210,6 +210,11 @@ CSHARP_COMMON_ATTRS = dicts.add(
             # that is generated by a dotnet binary
             cfg = default_transition,
         ),
+        "allow_unsafe": attr.bool(
+            doc = "Allow to compile unsafe code. It true, /unsafe is pass to the compiler.",
```
Comment (purkhusid) [inline dotnet/private/rules/common/attrs.bzl:214]:
```suggestion
            doc = "Allow compiling unsafe code. It true, /unsafe is passed to the compiler.",
```

**Comment by purkhusid** [comment]
I renamed the option from `allow_unsafe` to just `allow_unsafe_blocks` to reflect the MSBuild option.
Thanks for your contribution!

**Review by purkhusid — APPROVED** [review]

**Pull request merged**